src/main/java/at/hannibal2/skyhanni/utils/InventoryUtils.kt

```diff
@@ -40,7 +43,14 @@ object InventoryUtils {
     fun countItemsInLowerInventory(predicate: (ItemStack) -> Boolean) =
         getItemsInOwnInventory().filter { predicate(it) }.sumOf { it.stackSize }
 
-    fun getArmor(): Array<ItemStack?> = Minecraft.getMinecraft().thePlayer.inventory.armorInventory
+    fun getArmor(dataAgeLimit: Int): Array<ItemStack?> {
+        if (lastArmorCheck.passedSince() > dataAgeLimit.milliseconds) {
+            wornArmorNames = Minecraft.getMinecraft().thePlayer.inventory.armorInventory
+            lastArmorCheck = SimpleTimeMark.now()
+        }
```

Comment:
This does not look to me like proper delay on the armor. I can only guess the functionality of this method (which makes it bad to begin with and deserves to be commented on), but I will assume this is meant to counteract the fact that the server takes about a second to calculate whether or not you are wearing an armor set. In this case the delay might be offset from when you actually switch armor, because the timer is not aligned with any armor switch, just with the last method call.